### ISSUE TITLE
wrpl unpacker is outdated

### ISSUE THREAD

**Issue by Yay5379** [klensy/wt-tools#72]
The wrpl unpacker is not compatible with replays from newer versions of war thunder. (This is most likely due to the addition of the nuclear bomb)

**Comment by OshidaBCF** [comment]
wrpl unpacker hasn't been updated since 3 years, the game evolved a lot in that timestamp, anything would make the replays uncompatible, as just a simple turret speed change would make it uncompatible.
By default, a replay only works in it's version, not another one.

**Comment by kotiq** [comment]
What do you find useful inside the replay? Do the versions before the format change have value for you? 
Now the replay contains blocks whose structure differs from the previous one: `[Header] [DataSize] [Data]`
I will have to recognize it as the data flow progresses.
It is easier for me to support only the current version, so that the parser is not cluttered with dead branches.

**Comment by OshidaBCF** [comment]
There isn't any real useful info in replay...

So idk what it could be used for

**Comment by kotiq** [comment]
Yes, the content of the blocks has not changed. I guessed many fields from the header. From the useful there is the version, the size of the first block and the offset of the last block. Thus, for all 3 blocks, you can calculate the size for a given header size of `0x4c4`. The rest of them duplicate the fields from `replays.wdb` and the first block.

**Comment by TaLajans** [comment]
Hello! I also have some interest in having .wrpl extract working and I do hope that server replay can hold a bit more information about the match.

### Problem
This mission end window is displayed only once in a game right after you leave a match or it ended.

![init](https://user-images.githubusercontent.com/92247810/136696486-15a94eab-32cd-4ff0-86d4-ca0ac6808607.jpg)

From this window you can get a lot of information about each of the vehicle you used in this match.
- Time spent.
- Total critical hits
- Total hits
- Total kills
- Total caps
- some other.

![time](https://user-images.githubusercontent.com/92247810/136696481-c45f8600-122c-4766-ab98-1b2a749756a7.jpg)
![crit](https://user-images.githubusercontent.com/92247810/136696483-bdc67930-2436-4800-9e7e-42a5aed1b66b.jpg)
![groundkills](https://user-images.githubusercontent.com/92247810/136696484-aa11dcb5-f62f-462b-84bd-1b6f9b2515af.jpg)
![hits](https://user-images.githubusercontent.com/92247810/136696485-aeefbd41-77a3-416c-8ffc-6b234ed90573.jpg)

### Possible outcome???

Would it be possible to parse/extract server replays just to get key server data and ignore all other information that could change during next update?

- Timestamp of user spawn a vehicle. 
- Timestamp of user scoring a kill/hit/crit hit in specific vehicle.
- Timestamp of user successfully capping a point.
- Timestamp of user dying in vehicle

Server replays are split into multiple parts. They are listed at https://warthunder.com/en/tournament/replay/ and this is an example replay: https://warthunder.com/en/tournament/replay/type/154177510472539583

![replay](https://user-images.githubusercontent.com/92247810/136696526-cea6088b-fdf2-4fd4-a014-56c7580c2f16.jpg)

**Comment by Yay5379** [comment]
Random Update (It's been 2 years): Whenever I try to extract a replay I noticed that the error message is always construct.core.ConstError: parsing expected b'\x00BBF' but parsed b'kill'. I also noticed that replays that do not have a nuke are able to be unpacked. This leads me to theorize that the nuke causes some sort of quirk in the replay system.

**Comment by Yay5379** [comment]
Opening a new issue with more details about what the problem might be.

**Comment by OshidaBCF** [comment]
> This leads me to theorize that the nuke causes some sort of quirk in the replay system.

This would make a lot of sense.
